app_flutter/lib/agent_dashboard_page.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_progress_button/flutter_progress_button.dart';
+import 'package:provider/provider.dart';
+
+import 'agent_list.dart';
+import 'navigation_drawer.dart';
+import 'service/google_authentication.dart';
+import 'sign_in_button.dart';
+import 'state/agent.dart';
+
+/// [AgentDashboardPage] parent widget that manages the state of the dashboard.
+class AgentDashboardPage extends StatefulWidget {
+  AgentDashboardPage({
+    AgentState agentState,
+    GoogleSignInService signInService,
+    @visibleForTesting this.agentFilter,
+  }) : agentState = agentState ?? AgentState(authServiceValue: signInService);
+
+  static const String routeName = '/agents';
+
+  final AgentState agentState;
+
+  final String agentFilter;
+
+  @visibleForTesting
+  static const Duration errorSnackbarDuration = Duration(seconds: 8);
+  @override
+  _AgentDashboardPageState createState() => _AgentDashboardPageState();
+}
+
+class _AgentDashboardPageState extends State<AgentDashboardPage> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  AgentState agentState;
+
+  @override
+  void initState() {
+    super.initState();
+
+    widget.agentState.startFetchingStateUpdates();
+
+    widget.agentState.errors.addListener(_showErrorSnackbar);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    agentState = widget.agentState;
```

Comment:
Why are you pulling it off of the widget here?

Reply:
Good catch, moved to `initState()`.